packages/pluggableWidgets/line-chart-web/rollup.config.js

```diff
@@ -0,0 +1,93 @@
+import { join } from "path";
+import replace from "rollup-plugin-re";
+import { nodeResolve } from "@rollup/plugin-node-resolve";
+import { getBabelInputPlugin, getBabelOutputPlugin } from "@rollup/plugin-babel";
+import commonjs from "@rollup/plugin-commonjs";
+import { terser } from "rollup-plugin-terser";
+
+export default args => {
+    const production = Boolean(args.configProduction);
+    const outDir = join(__dirname, "dist/tmp/widgets/com/mendix/shared");
+    const result = args.configDefaultConfig;
+    const [jsConfig, mJsConfig] = result;
+
+    // We define the new library externals for the entry points
+    const libraryExternals = [/^react-plotly\.js($|\/)/, /(^|\/)shared\/api.js$/];
+
+    // We force the original configuration to replace the library with the local implementation
+    // Configuration for the main entry point for the client
+    jsConfig.plugins.splice(
+        -1,
+        0,
+        replace({
+            replaces: {
+                // It considers the relative path from the file which the library is being imported
+                "react-plotly.js": "../../../../shared/api.js"
+            }
+        })
+    );
+    // Configuration for Modern Client (mJS)
+    mJsConfig.plugins.splice(
+        -1,
+        0,
+        replace({
+            replaces: {
+                "react-plotly.js": "../../../../shared/api.js"
+            }
+        })
+    );
+
+    const externals = jsConfig.external;
+
+    // We force the externals to contain the library + the just built api file
+    jsConfig.external = [...externals, ...libraryExternals];
+    mJsConfig.external = [...externals, ...libraryExternals];
```

Comment:
This removes the library from the original bundle right?

Reply:
Yeap, the newly bundled lib and the original (api.js)